src/figuresClasses.ts

```diff
@@ -2,10 +1,83 @@
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = `red` | `green` | `blue`;
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a < 1 || this.b < 1 || this.c < 1) {
+      throw new Error('Sides can`t be less than 1');
```

Comment:
[CODE STYLE] - When throwing an error, include a more detailed description in the error message. Suggested message: 'Triangle side lengths must be greater than 0.'